flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2049,10 +2085,11 @@ static void genOMPDispatch(lower::AbstractConverter &converter,
     genParallelOp(converter, symTable, semaCtx, eval, loc, queue, item);
     break;
   case llvm::omp::Directive::OMPD_section:
-    genSectionOp(converter, symTable, semaCtx, eval, loc, queue, item);
+    // Lowered in the enclosing genSectionsOp.
```

Comment:
Maybe add an `llvm_unreachable()` here and in the 'sections' case?

Reply:
In the sections case the code is actually reachable and needs to do nothing. I added a comment there to explain that.